### PR TITLE
Ice-shelf FMAs

### PR DESCRIPTION
**Pull request by alex-huth**
Fixes to preserve rotational symmetry when using Fused-multiply-adds. Tested in ice-ocean mode and ice-shelf-only mode using ISOMIP and MISOMIP+, respectively

### CODE REVIEW

File: src/ice_shelf/MOM_ice_shelf_dynamics.F90 [hunk 1]
```diff
@@ -282,9 +282,9 @@ function quad_area (X, Y)
     !  |   |
     !  1 - 2
 
-  p2 = (X(4)-X(1))**2 + (Y(4)-Y(1))**2 ; q2 = (X(3)-X(2))**2 + (Y(3)-Y(2))**2
```
Comment (Hallberg-NOAA) [inline src/ice_shelf/MOM_ice_shelf_dynamics.F90:285]: If x and y are corresponding directional variables, we might also need to add parentheses around the squares in these 4 expressions, as well as the newly added ones around the whole expressions.

File: src/ice_shelf/MOM_ice_shelf_dynamics.F90 [hunk 2]
```diff
@@ -3358,10 +3358,10 @@ subroutine bilinear_shape_functions (X, Y, Phi, area)
 
   do qpoint=1,4
 
-    a = -X(1)*(1-yquad(qpoint)) + X(2)*(1-yquad(qpoint)) - X(3)*yquad(qpoint) + X(4)*yquad(qpoint) ! d(x)/d(x*)
```
Comment (Hallberg-NOAA) [inline src/ice_shelf/MOM_ice_shelf_dynamics.F90:3361]: The changes on these lines are mathematically equivalent to what was here before, but in specifying the correct order of arithmetic to give rotational symmetry, they are likely to change answers for any cases that use a dynamic ice sheet.  That being said, I do agree that these precise changes are both correct an necessary.

**Review by Hallberg-NOAA — APPROVED** [review]
These changes all look correct to me, and I agree with them.

  It should be noted that the mathematically equivalent changes on lines 3361 to 3364 of MOM_ice_shelf_dynamics.F90 are likely to change answers for cases that use a dynamic ice sheet, regardless of whether FMAs are being used.

File: src/ice_shelf/MOM_ice_shelf_dynamics.F90 (hunk 2 above)
Comment (Hallberg-NOAA) [reply src/ice_shelf/MOM_ice_shelf_dynamics.F90:3361]: I do not believe that there are any widely used cases with a dynamic ice sheet that would be adversely impacted by these specific changes, so I do not think that we need to add an answer date flag here to recover the previous (indeterminate) order of arithmetic.

**Comment by marshallward** [comment]
Gaea regression: https://gitlab.gfdl.noaa.gov/ogrp/mom6ci/MOM6/-/pipelines/25072 :heavy_check_mark:

**Pull request merged**